source/slang/emit.cpp

```diff
@@ -5995,7 +5995,14 @@ struct EmitVisitor
             }
         }
         
-        emit(") buffer ");
+        emit(") ");
+
+        if (as<IRHLSLStructuredBufferType>(structuredBufferType))
```

Comment:
Maybe a comment here (and in the case below) to indicate the rationale for why we only check for these specific cases before emitting `readonly`?